workbench-core/example/infrastructure/integration-tests/tests/dynamicAuthorization/identityPermissions.test.ts

```diff
@@ -275,10 +254,279 @@ describe('dynamic authorization identity permission integration tests ', () => {
       const exceededIdentityPermissions = Array(101).fill(identityPermissions[0]);
       await expect(
         adminSession.resources.identityPermissions.delete({
-          identityPermissions: exceededIdentityPermissions,
-          authenticatedUser: mockUser
+          identityPermissions: exceededIdentityPermissions
         })
       ).rejects.toThrow(new HttpError(429, {}));
     });
   });
+  describe('isAuthorizedOnRoute', () => {
+    let mockIdentityPermissions: IdentityPermission[];
+    let subjectType: string;
+    let subjectId: string;
+    let parentId: string;
+    let randomTextGenerator: RandomTextGenerator;
+    let newAdminSession: ClientSession;
+    let groupId: string;
+    beforeAll(async () => {
+      const { data } = await adminSession.resources.groups.create();
+      groupId = data.groupId;
+      randomTextGenerator = new RandomTextGenerator('isAuthorizedOnSubject');
+      randomTextGenerator = new RandomTextGenerator('isAuthorizedOnSubject');
+      subjectType = 'sampleResource';
+      subjectId = randomTextGenerator.getFakeText('sampleSubjectId');
+      parentId = randomTextGenerator.getFakeText('sampleParentId');
+      mockIdentityPermissions = [
+        {
+          subjectType,
+          subjectId,
+          identityId: groupId,
+          identityType: 'GROUP',
+          action: 'UPDATE',
+          effect: 'ALLOW',
+          conditions: {
+            parentId: { $eq: parentId }
+          }
+        },
+        {
+          subjectType,
+          subjectId: '*',
+          identityId: groupId,
+          identityType: 'GROUP',
+          action: 'READ',
+          effect: 'ALLOW',
+          conditions: {
+            parentId: { $eq: parentId }
+          }
+        }
+      ];
+      await adminSession.resources.identityPermissions.create(
+        {
+          identityPermissions: mockIdentityPermissions
+        },
+        false
+      );
+      await adminSession.resources.groups
+        .group(groupId)
+        .addUser({ userId: adminSession.getSettings().get('rootUserId') });
+      newAdminSession = await setup.getDefaultAdminSession(true);
+    });
+    //These routes are defined in dynamicRouteConfig and sampleRoutes
+    test('is authorized on to PUT /parentResource/:parentId/resource/:resourceId with correct permissions', async () => {
```

Comment:
do we need a negative test for PUT as well for `subjectID`s that you aren't allowed to update?